### ISSUE TITLE
Not showing any error

### ISSUE THREAD

**Issue by wellyal** [SublimeLinter/SublimeLinter#782]
I wasnt able to make it work. I followed the instructions on the [documentation](http://sublimelinter.readthedocs.io/en/latest/installation.html) installed through package install, restarted sublime but it does not show the lint errors. On console loads everything ok like:

System: Mac OS EL Capitan v10.11.6
Sublime 3

```
reloading plugin SublimeLinter.commands
reloading plugin SublimeLinter.sublimelinter
plugins loaded
```

eslint working just fine though
![image](https://user-images.githubusercontent.com/3047016/34778397-62b68d2a-f604-11e7-8374-27123f60db0f.png)

**Comment by FichteFoll** [comment]
You should enable debug logging for SublimeLinter, restart ST and post the entire console's contents here.

**Comment by wellyal** [comment]
```
...
reloading plugin SublimeLinter.commands
reloading plugin SublimeLinter.sublimelinter
plugins loaded
SublimeLinter: debug mode: on 
SublimeLinter: temp directory: /var/folders/r5/mq06kxp92njczjyb1qv56qzc0000gp/T/SublimeLinter3-wellyngton.leitao 
Package Control: Skipping automatic upgrade, last run at 2018-01-10 12:34:22, next run at 2018-01-10 13:34:22 or after
environment variables loaded using: /bin/zsh -l
```

**Comment by thezimmee** [comment]
SublimeLinter also broke for me when I upgraded to `v4.0.0-rc.9.4`, where it was working prior to the upgrade (with the exception of known issues like the inline diff, .sublime-color-theme, etc., all which I believe `v4.0.0-rc.9.4` is supposed to address). Now the only linter that "kinda" works is annotations. No errors show for other linters; annotations shows the error but doesn't highlight anything as it used to. I saw this error in the console:

```
Traceback (most recent call last):
  File "/Volumes/Home/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/util.py", line 264, in generate_color_scheme_async
  File "./python3.3/xml/etree/ElementTree.py", line 1356, in XML
xml.etree.ElementTree.ParseError: not well-formed (invalid token): line 1, column 0
```

And this is the console output when starting up:

```
startup, version: 3156 osx x64 channel: dev
executable: /Applications/Sublime Text.app/Contents/MacOS/Sublime Text
working dir: /
packages path: /Volumes/Home/Library/Application Support/Sublime Text 3/Packages
state path: /Volumes/Home/Library/Application Support/Sublime Text 3/Local
zip path: /Applications/Sublime Text.app/Contents/MacOS/Packages
zip path: /Volumes/Home/Library/Application Support/Sublime Text 3/Installed Packages
ignored_packages: ["AutoFileName", "Color Highlighter", "Markdown", "Sublimerge 3", "Vintage"]
pre session restore time: 0.266407
using gpu buffer for window
loading dictionary Packages/Language - English/en_US.dic
startup time: 0.380774
...
plugins loaded
SublimeLinter: debug mode: on 
SublimeLinter: temp directory: /var/folders/06/7cjy1d5930343lrshpww5sdh0000gp/T/SublimeLinter3-zimmee 
SublimeLinter: computed PATH using /bin/zsh:
/usr/bin
/bin
/usr/sbin
/sbin
 
SublimeLinter: WARNING: mdl deactivated, cannot locate the gem 'mdl' 
FuzzyFilePath cached 44 files in /Volumes/Home/Projects/brikcss/tim
SublimeLinter: eslint: index.js ['/Volumes/Home/Projects/brikcss/tim/node_modules/.bin/eslint', '--format', 'compact', '--stdin', '--stdin-filename', '@'] 
SublimeLinter: eslint output:
env: node: No such file or directory 
SublimeLinter: No match for regex: ^.+?: line (?P<line>\d+), col (?P<col>\d+), (?:(?P<error>Error)|(?P<warning>Warning)) - (?P<message>.+) 
environment variables loaded using: /bin/zsh -l
SublimeLinter: annotations: index.js output:
line 1, col 26: @todo: Move this loop to the `compileBundle()` function? 
FuzzyFilePath cached 44 files in /Volumes/Home/Projects/brikcss/tim
Package Control: No updated packages
SublimeLinter: eslint: index.js ['/Volumes/Home/Projects/brikcss/tim/node_modules/.bin/eslint', '--format', 'compact', '--stdin', '--stdin-filename', '@'] 
SublimeLinter: eslint output:
env: node: No such file or directory 
SublimeLinter: No match for regex: ^.+?: line (?P<line>\d+), col (?P<col>\d+), (?:(?P<error>Error)|(?P<warning>Warning)) - (?P<message>.+) 
SublimeLinter: annotations: index.js output:
line 1, col 26: @todo: Move this loop to the `compileBundle()` function? 
SublimeLinter: eslint: index.js ['/Volumes/Home/Projects/brikcss/tim/node_modules/.bin/eslint', '--format', 'compact', '--stdin', '--stdin-filename', '@'] 
SublimeLinter: eslint output:
env: node: No such file or directory 
SublimeLinter: No match for regex: ^.+?: line (?P<line>\d+), col (?P<col>\d+), (?:(?P<error>Error)|(?P<warning>Warning)) - (?P<message>.+) 
SublimeLinter: annotations: index.js output:
line 1, col 26: @todo: Move this loop to the `compileBundle()` function? 
SublimeLinter: eslint: index.js ['/Volumes/Home/Projects/brikcss/tim/node_modules/.bin/eslint', '--format', 'compact', '--stdin', '--stdin-filename', '@'] 
SublimeLinter: eslint output:
env: node: No such file or directory 
SublimeLinter: No match for regex: ^.+?: line (?P<line>\d+), col (?P<col>\d+), (?:(?P<error>Error)|(?P<warning>Warning)) - (?P<message>.+) 
SublimeLinter: annotations: index.js output:
line 1, col 26: @todo: Move this loop to the `compileBundle()` function? 
SublimeLinter: eslint: index.js ['/Volumes/Home/Projects/brikcss/tim/node_modules/.bin/eslint', '--format', 'compact', '--stdin', '--stdin-filename', '@'] 
SublimeLinter: eslint output:
env: node: No such file or directory 
SublimeLinter: No match for regex: ^.+?: line (?P<line>\d+), col (?P<col>\d+), (?:(?P<error>Error)|(?P<warning>Warning)) - (?P<message>.+) 
SublimeLinter: annotations: index.js output:
line 1, col 26: @todo: Move this loop to the `compileBundle()` function? 
SublimeLinter: eslint: index.js ['/Volumes/Home/Projects/brikcss/tim/node_modules/.bin/eslint', '--format', 'compact', '--stdin', '--stdin-filename', '@'] 
SublimeLinter: eslint output:
env: node: No such file or directory 
SublimeLinter: No match for regex: ^.+?: line (?P<line>\d+), col (?P<col>\d+), (?:(?P<error>Error)|(?P<warning>Warning)) - (?P<message>.+) 
SublimeLinter: annotations: index.js output:
line 1, col 26: @todo: Move this loop to the `compileBundle()` function? 
SublimeLinter: eslint: index.js ['/Volumes/Home/Projects/brikcss/tim/node_modules/.bin/eslint', '--format', 'compact', '--stdin', '--stdin-filename', '@'] 
SublimeLinter: eslint output:
env: node: No such file or directory 
SublimeLinter: No match for regex: ^.+?: line (?P<line>\d+), col (?P<col>\d+), (?:(?P<error>Error)|(?P<warning>Warning)) - (?P<message>.+) 
SublimeLinter: annotations: index.js output:
line 1, col 26: @todo: Move this loop to the `compileBundle()` function? 
```

**Comment by FichteFoll** [comment]
Looks like the `node` executable can't be found in ST's PATH env variable.

**Comment by kaste** [comment]
@thezimmee @FichteFoll  This is not a good path for osx, is it?:

![image](https://user-images.githubusercontent.com/8558/34798019-aaa512ae-f65a-11e7-81da-2b897b98191a.png)

Is `'/Volumes/Home/Projects/brikcss/tim/node_modules/.bin/eslint' actually a good eslint executable?

**Comment by kaste** [comment]
@wellyal This isn't the complete log. Do you have https://github.com/SublimeLinter/SublimeLinter-eslint actually installed?

**Comment by thezimmee** [comment]
@kaste I believe that is the path for a _local_ installation of the eslint executable; so it would exist so long as the user has eslint installed in that project. But it's definitely not the global install path. Weird thing is, I actually have eslint installed in that project, so that path should work for me. But it is not. I do have SublimeLInter-eslint installed.

**Comment by kaste** [comment]
@thezimmee For me, the output is telling me, that it can't find `node` within ```/usr/bin
/bin
/usr/sbin
/sbin```

**Comment by kaste** [comment]
@thezimmee  You could try to patch 

https://github.com/SublimeLinter/SublimeLinter/blob/4ca6eac6bfb9ea9bfd19bffd8b86bacc0fec347e/lint/util.py#L238

Remove that `lrucache` line. And see, if the path gets better over time.

**Comment by thezimmee** [comment]
My node is installed to `/usr/local/bin/node`, so that would explain it. I don't see anywhere in SublimeLinter to tell it where to look for node... I just tried your patch, thank you!! It does appear to be working -- eslint is actually showing errors now -- but how does this get fixed permanently? This worked prior to the latest update, on my same setup/system, nothing has changed on my machine, so I wonder why suddenly it can't find node? I also wonder if this fails for those who use [nvm](https://github.com/creationix/nvm) (I believe nvm has different locations where it installs node to)?

**Comment by FichteFoll** [comment]
What shell do you use? (zsh probably)
Within that shell, what does `echo $PATH` print?

As @kaste rightfully noticed, this seems to be a problem with trying to infer your environment's PATH from your shell in a (more or less) interactive session, because macOS is weird that way and applications launched from the dock do not get the environment of the login shell, which is why it has do be inferred manually. (And many tools get installed to `/usr/local/bin`, which is not included in the PATH by default.)

And finally, please, *please*, include the *entire* console log because there is a lot of useful information in there beyond just lines containing "SublimeLinter".

**Comment by thezimmee** [comment]
> What shell do you use? (zsh probably)

Correct, zsh.

> Within that shell, what does echo $PATH print?

```shell
/Library/Frameworks/Python.framework/Versions/3.6/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/bin:/usr/sbin:/sbin:/Volumes/Home/go-projects/bin:/usr/local/opt/go/libexec/bin:/Applications/Araxis Merge.app/Contents/Utilities print
```

> And finally, please, please, include the entire console log because there is a lot of useful information in there beyond just lines containing "SublimeLinter".

All I removed from the original output above were the notifications of plugins loading. But here is a fresh set of output from SublimeLinter in a non-working state, without any output removed, in case it helps:

```
startup, version: 3156 osx x64 channel: dev
executable: /Applications/Sublime Text.app/Contents/MacOS/Sublime Text
working dir: /
packages path: /Volumes/Home/Library/Application Support/Sublime Text 3/Packages
state path: /Volumes/Home/Library/Application Support/Sublime Text 3/Local
zip path: /Applications/Sublime Text.app/Contents/MacOS/Packages
zip path: /Volumes/Home/Library/Application Support/Sublime Text 3/Installed Packages
ignored_packages: ["AutoFileName", "Color Highlighter", "FileSystem Autocompletion", "Markdown", "Sublimerge 3", "Vintage"]
pre session restore time: 0.217086
using gpu buffer for window
loading dictionary Packages/Language - English/en_US.dic
startup time: 0.340844
Error loading scope:source.jade: Unable to find syntax file for scope source.jade in Packages/MarkdownHighlighting/Markdown.sublime-syntax
Error loading scope:source.nginx: Unable to find syntax file for scope source.nginx in Packages/MarkdownHighlighting/Markdown.sublime-syntax
Error loading scope:source.scss: Unable to find syntax file for scope source.scss in Packages/MarkdownHighlighting/Markdown.sublime-syntax
Error loading scope:source.css.less: Unable to find syntax file for scope source.css.less in Packages/MarkdownHighlighting/Markdown.sublime-syntax
Error loading scope:source.ejs: Unable to find syntax file for scope source.ejs in Packages/MarkdownHighlighting/Markdown.sublime-syntax
Error loading scope:source.swift: Unable to find syntax file for scope source.swift in Packages/MarkdownHighlighting/Markdown.sublime-syntax
Error loading scope:source.pgsql: Unable to find syntax file for scope source.pgsql in Packages/MarkdownHighlighting/Markdown.sublime-syntax
Error loading scope:source.graphql: Unable to find syntax file for scope source.graphql in Packages/MarkdownHighlighting/Markdown.sublime-syntax
Error loading scope:source.coffee: Unable to find syntax file for scope source.coffee in Packages/MarkdownHighlighting/Markdown.sublime-syntax
Error loading scope:source.stylus: Unable to find syntax file for scope source.stylus in Packages/MarkdownHighlighting/Markdown.sublime-syntax
Error loading scope:text.html.handlebars: Unable to find syntax file for scope text.html.handlebars in Packages/MarkdownHighlighting/Markdown.sublime-syntax
Error loading scope:text.html.swig: Unable to find syntax file for scope text.html.swig in Packages/MarkdownHighlighting/Markdown.sublime-syntax
Error loading scope:source.livescript: Unable to find syntax file for scope source.livescript in Packages/MarkdownHighlighting/Markdown.sublime-syntax
Error loading scope:source.julia: Unable to find syntax file for scope source.julia in Packages/MarkdownHighlighting/Markdown.sublime-syntax
reloading plugin Default.arithmetic
reloading plugin Default.auto_indent_tag
reloading plugin Default.block
reloading plugin Default.colors
reloading plugin Default.comment
reloading plugin Default.convert_color_scheme
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.install_package_control
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.quick_panel
reloading plugin Default.rename
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.settings
reloading plugin Default.show_scope_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin Default.ui
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin ShellScript.ShellScript
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.01-pygments
reloading plugin 0_package_control_loader.50-markupsafe
reloading plugin 0_package_control_loader.50-pymdownx
reloading plugin 0_package_control_loader.50-python-markdown
reloading plugin 0_package_control_loader.50-pyyaml
reloading plugin 0_package_control_loader.51-python-jinja2
reloading plugin 0_package_control_loader.55-jsonschema
reloading plugin 0_package_control_loader.55-mdpopups
reloading plugin A File Icon.A File Icon
reloading plugin AdvancedNewFile.AdvancedNewFile
reloading plugin ColorHelper.color_helper
reloading plugin ColorHelper.color_helper_insert
reloading plugin ColorHelper.color_helper_picker
reloading plugin ColorHelper.color_helper_util
reloading plugin ColorHelper.multiconf
reloading plugin ColorHelper.support
reloading plugin Compare Side-By-Side.sbs_compare
reloading plugin CSS Color Converter.colorconvert
reloading plugin DocBlockr.jsdocs
reloading plugin EditorConfig.EditorConfig
reloading plugin ExpandRegion.__init__
reloading plugin ExpandRegion.expand_region_handler
reloading plugin ExpandRegion.expand_to_indent
reloading plugin ExpandRegion.expand_to_line
reloading plugin ExpandRegion.expand_to_quotes
reloading plugin ExpandRegion.expand_to_regex_set
reloading plugin ExpandRegion.expand_to_semantic_unit
reloading plugin ExpandRegion.expand_to_subword
reloading plugin ExpandRegion.expand_to_symbols
reloading plugin ExpandRegion.expand_to_word
reloading plugin ExpandRegion.expand_to_word_with_dots
reloading plugin ExpandRegion.expand_to_xml_node
reloading plugin ExpandRegion.ExpandRegion
reloading plugin ExpandRegion.html
reloading plugin ExpandRegion.javascript
reloading plugin ExpandRegion.latex
reloading plugin ExpandRegion.python
reloading plugin ExpandRegion.utils
reloading plugin FindKeyConflicts.find_key_conflicts
reloading plugin Focus File on Sidebar.FocusFileOnSidebar
reloading plugin Focus Last Tab.focus_last_tab
reloading plugin FuzzyFileNav.fuzzy_file_nav
reloading plugin FuzzyFileNav.multiconf
reloading plugin FuzzyFileNav.notify
reloading plugin FuzzyFilePath.command_goto_file
reloading plugin FuzzyFilePath.command_insert_path
reloading plugin FuzzyFilePath.command_rebuild_cache
reloading plugin FuzzyFilePath.command_replace_region
reloading plugin FuzzyFilePath.command_show_context
reloading plugin FuzzyFilePath.command_show_current_settings
reloading plugin FuzzyFilePath.command_show_info
reloading plugin FuzzyFilePath.completion
reloading plugin FuzzyFilePath.controller
reloading plugin FuzzyFilePath.current_state
reloading plugin FuzzyFilePath.expression
reloading plugin FuzzyFilePath.ProjectListener
reloading plugin FuzzyFilePath.query
reloading plugin FuzzyFilePath.QueryCompletionListener
reloading plugin FuzzyFilePath.TestRunner
reloading plugin FuzzyFilePath.ViewListener
reloading plugin GitGutter.plugin
reloading plugin GitSavvy.__init__
reloading plugin GitSavvy.git_savvy
reloading plugin Hex to HSL Color Converter.Hex-to-HSL-color
reloading plugin Hex to HSL Color Converter.RGB-to-HSL-color
reloading plugin Inc-Dec-Value.inc_dec_value
reloading plugin KeybindingHelper.command_assistant
reloading plugin KeybindingHelper.paste_keybinding
reloading plugin KeybindingHelper.show_keys
reloading plugin KeyboardSelection.KeyboardSelection
reloading plugin MarkdownEditing.bootstrap
reloading plugin MarkdownEditing.custom_find_under_expand
reloading plugin MarkdownEditing.decide_title
reloading plugin MarkdownEditing.distraction_free_mode
reloading plugin MarkdownEditing.folding
reloading plugin MarkdownEditing.footnotes
reloading plugin MarkdownEditing.indent_list_item
reloading plugin MarkdownEditing.indent_list_multiitem
reloading plugin MarkdownEditing.lint
reloading plugin MarkdownEditing.mdeutils
reloading plugin MarkdownEditing.numbered_list
reloading plugin MarkdownEditing.quote_indenting
reloading plugin MarkdownEditing.references
reloading plugin MarkdownEditing.switch_list_bullet_type
reloading plugin MarkdownEditing.underlined_headers
reloading plugin MarkdownTOC.MarkdownTOC
reloading plugin MaxPane.layouts_doc
reloading plugin MaxPane.max_pane
reloading plugin MoveTab.move_tab
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
reloading plugin PackageResourceViewer.package_resource_viewer
reloading plugin PackageResourceViewer.package_resources
reloading plugin ProjectManager.json_file
reloading plugin ProjectManager.project_manager
reloading plugin Resize Group with Keyboard.resize_active_group
reloading plugin ScopeHunter.scope_hunter
reloading plugin ScopeHunter.scope_hunter_notify
reloading plugin ScopeHunter.support
reloading plugin ShellCommand.OsShell
reloading plugin ShellCommand.ShellCommand
reloading plugin ShellCommand.SublimeHelper
reloading plugin ShellCommand.VariableSubstitution
reloading plugin SideBarEnhancements.SideBar
reloading plugin SideBarEnhancements.SideBarAPI
reloading plugin SideBarEnhancements.SideBarDefaultDisable
reloading plugin SublimeLinter-annotations.linter
reloading plugin SublimeLinter-contrib-mdl.linter
reloading plugin SublimeLinter-eslint.linter
reloading plugin SublimeLinter-html-tidy.linter
reloading plugin SublimeLinter-json.linter
reloading plugin SublimeLinter.busy_indicator_view
reloading plugin SublimeLinter.commands
reloading plugin SublimeLinter.status_bar_view
reloading plugin SublimeLinter.sublime_linter
reloading plugin Table Editor.table_base
reloading plugin Table Editor.table_border_syntax
reloading plugin Table Editor.table_emacs_org_mode_syntax
reloading plugin Table Editor.table_lib
reloading plugin Table Editor.table_lib_test
reloading plugin Table Editor.table_line_parser
reloading plugin Table Editor.table_multi_markdown_syntax
reloading plugin Table Editor.table_pandoc_syntax
reloading plugin Table Editor.table_plugin
reloading plugin Table Editor.table_plugin_test
reloading plugin Table Editor.table_re_structured_text_syntax
reloading plugin Table Editor.table_simple_syntax
reloading plugin Table Editor.table_textile_syntax
reloading plugin Table Editor.widechar_support
reloading plugin ThemeTweaker.support
reloading plugin ThemeTweaker.theme_tweaker
reloading plugin BetterFindBuffer.find_results
reloading plugin Codecs33.Codecs33
reloading plugin Cycle Setting.cycle-setting-command
reloading plugin FocusActiveView.focus-active-view
reloading plugin JsPrettier.JsPrettier
reloading plugin Open Command Palette.open-command-palette-command
reloading plugin Run Multiple Commands.run-multiple-commands
reloading plugin Save Active Theme.save-active-theme
reloading plugin SublimeLinter-contrib-stylelint.linter
plugins loaded
SublimeLinter: debug mode: on 
SublimeLinter: temp directory: /var/folders/06/7cjy1d5930343lrshpww5sdh0000gp/T/SublimeLinter3-zimmee 
SublimeLinter: computed PATH using /bin/zsh:
/usr/bin
/bin
/usr/sbin
/sbin
 
SublimeLinter: WARNING: mdl deactivated, cannot locate the gem 'mdl' 
FFP	 Settings no folder settings found ../../../../../../../Projects/brikcss/tim

FuzzyFilePath cached 50 files in /Volumes/Home/Projects/brikcss/tim
FuzzyFilePath cached 50 files in /Volumes/Home/Projects/brikcss/tim
SublimeLinter: eslint: index.js ['/Volumes/Home/Projects/brikcss/tim/node_modules/.bin/eslint', '--format', 'compact', '--stdin', '--stdin-filename', '@'] 
SublimeLinter: eslint output:
env: node: No such file or directory 
SublimeLinter: No match for regex: ^.+?: line (?P<line>\d+), col (?P<col>\d+), (?:(?P<error>Error)|(?P<warning>Warning)) - (?P<message>.+) 
environment variables loaded using: /bin/zsh -l
SublimeLinter: annotations: index.js output:
line 1, col 4: @todo: Move this loop to the `compileBundle()` function? 
Package Control: Skipping automatic upgrade, last run at 2018-01-10 16:06:37, next run at 2018-01-10 17:06:37 or after
FFP	 CurrentState Abort -- view already updated
FFP	 CurrentState Abort -- view already updated
FFP	 CurrentState Abort -- view already updated
FFP	 Settings no folder settings found ../../../../../../../Projects/brikcss/tim

SublimeLinter: eslint: .brikrc.js ['/Volumes/Home/Projects/brikcss/tim/node_modules/.bin/eslint', '--format', 'compact', '--stdin', '--stdin-filename', '@'] 
SublimeLinter: eslint output:
env: node: No such file or directory 
SublimeLinter: No match for regex: ^.+?: line (?P<line>\d+), col (?P<col>\d+), (?:(?P<error>Error)|(?P<warning>Warning)) - (?P<message>.+) 
```

**Comment by FichteFoll** [comment]
Thank you.

I see that the internal env resolver runs the shell with the `-l` parameter to produce a login shell. I suspect that this is what's missing in the current SL code, though I'd like to confirm that.

```
environment variables loaded using: /bin/zsh -l
```

In the console, please run the following commands:

```py
import os; os.environ['PATH']
import subprocess; subprocess.check_output(["/bin/zsh", "-c", "echo $PATH"])
subprocess.check_output(["/bin/zsh", "-l", "-c", "echo $PATH"])
```

**Comment by thezimmee** [comment]
> In the console, please run the following commands and paste their outputs:

```
>>> import os; os.environ['PATH']
'/Library/Frameworks/Python.framework/Versions/3.6/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/bin:/usr/sbin:/sbin'
>>> import subprocess; subprocess.check_output(["/bin/zsh", "-c", "echo $PATH"])
b'/Library/Frameworks/Python.framework/Versions/3.6/bin:/usr/local/bin:/usr/local/sbin:/Library/Frameworks/Python.framework/Versions/3.6/bin:/usr/bin:/bin:/usr/sbin:/sbin\n'
>>> subprocess.check_output(["/bin/zsh", "-l", "-c", "echo $PATH"])
b'/Library/Frameworks/Python.framework/Versions/3.6/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Frameworks/Python.framework/Versions/3.6/bin\n'
```

> (I still wonder why it worked previously, though.)

That's what I wonder as well. It worked previously with incorrect filepaths, which were fixed in [this bug](https://github.com/SublimeLinter/SublimeLinter/issues/760). Two releases later and here we are.

**Comment by braver** [comment]
On a Mac, if you start ST in a way that it immediately shows a file (e.g. because you previously quit ST with files open), SublimeLinter will run before your environment is fully loaded. Previously SublimeLinter was quite slow to start and this didn’t occur very often (although it’s probably still the cause behind several bug reports). Also, SL will not try to run linters until you open a file, so if you open a blank ST instance, give it a second to set your PATH, and then open a file, it usually works just fine.

**Comment by FichteFoll** [comment]
Also, I believe we should perform some measurements on the path finding that is being cached. Kind of annoying that you have to restart ST when you add things to the path for example, and sometimes plugins even change ST's runtime vars. There is an issue for a virtualenv switching package for example.

**Comment by thezimmee** [comment]
> so if you open a blank ST instance, give it a second to set your PATH, and then open a file, it usually works just fine.

@braver It's true that I usually open Sublime with a bunch of files already open. So I tried closing all my files, restarted a blank ST instance, waited, opened a file and linted. No luck. Same issue as before.

**Comment by kaste** [comment]
@braver This is not true. `RubyLinter` is not lazy. 

* The fix here is to remove the `lrucache`. The lrucache was there bc previously `create_environment` did invoke a shell which is slow. Now we just do a `dict.update` here, which is fast enough so we don't need a cache here. Then we wouldn't cache wrong results, and users would get a working SL *eventually*. [This is not true for `RubyLinter` bc it is eager, and disables itself. You can see this above `SublimeLinter: WARNING: mdl deactivated, cannot locate the gem 'mdl'`]

* If we could tell a wrong PATH from a good one, we could abort an ongoing lint, and re-queue it. E.g. if the env PATH only contains 4 items (this is what I saw here and in other issues) it is known to be bad.

* However, we could tell users also to modify their `paths` settings in the SublimeLinter config.

**Comment by skyesofrock** [comment]
Just wanted to add that the new RC of ST4 also broke the installation for me. Reinstalled several times, but even the command palette options don't appear. Downgraded to the latest stable version and it's working fine.

**Comment by braver** [comment]
> The fix here is to remove the lrucache. 

Sounds good, let's drop it.

> If we could tell a wrong PATH from a good one, we could abort an ongoing lint, and re-queue it. 

In https://github.com/SublimeTextIssues/Core/issues/1877 Will mentions there may be a hook in upcoming ST versions that will allow us to restart linters after the shell environment has been loaded.

**Comment by kaste** [comment]
self-assign

**Comment by kaste** [comment]
#786 for the hotfix

**Comment by kaste** [comment]
@thezimmee kind of highjacked this issue. 😄  @wellyal Are you still here?

**Comment by thezimmee** [comment]
> @thezimmee kind of highjacked this issue. 😄 

My apologies. Didn't intend to take away from OP's issue, just seemed like a same or similar issue. Thanks to all for the help.

**Comment by braver** [comment]
#786 definitely fixes this for me. @wellyal if you still have problems let me know!

**Comment by kaste** [comment]
@thezimmee annotations should work again
